core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -4033,8 +4033,13 @@ public Object convertDataFromString( String pol, ValueMetaInterface convertMeta,
     Boolean isEmptyAndNullDiffer = convertStringToBoolean(
         Const.NVL( System.getProperty( Const.KETTLE_EMPTY_STRING_DIFFERS_FROM_NULL, "N" ), "N" ) );
 
-    if ( pol == null && isStringValue && isEmptyAndNullDiffer ) {
-      pol = Const.NULL_STRING;
+    Boolean normalizeNullStringToEmpty = !convertStringToBoolean(
+      Const.NVL( System.getProperty( Const.KETTLE_DO_NOT_NORMALIZE_NULL_STRING_TO_EMPTY, "N" ), "N" ) );
+
+    if ( normalizeNullStringToEmpty ) {
+      if ( pol == null && isStringValue && isEmptyAndNullDiffer ) {
```

Comment:
why not if ( normalizeNullStringToEmpty && pol == null && isStringValue && isEmptyAndNullDiffer ), basically just "merging" the two ifs.

Reply:
Just for the sake of clarity, so anyone will be able to tell what this change was all about in the not so distant future.